File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string code) {
    string answer = "";
    bool mode = false;
    for(int i = 0 ; i < code.size() ; i++ ){
        if(code[i]=='1')
            mode = !mode;
        else {
            if(i%2==0 && mode == false)
                answer.push_back(code[i]);
            if(i%2==1 && mode == true)
                answer.push_back(code[i]);
        }

    }
    
    if(answer == "") return "EMPTY";
    return answer;
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.swift
import Foundation

func solution(_ code:String) -> String {
    var result: Array<Character> = []
    var arr = Array(code)
    var mode = 0

    for i in 0..<arr.count {
        if arr[i] == "1" { // 모드 변경
            mode = mode == 0 ? 1 : 0
            continue
        }

        if mode == 0 && i % 2 == 0 {
            result.append(arr[i])
        } else if mode == 1 && i % 2 == 1 {
            result.append(arr[i])
        }
    }

    return result.isEmpty ? "EMPTY" : String(result)
}